cloudfoundry/resource_cf_app.go

```diff
@@ -639,7 +639,21 @@ func resourceAppUpdate(d *schema.ResourceData, meta interface{}) (err error) {
 		if err = am.UploadApp(app, appPath, addContent); err != nil {
 			return err
 		}
-		restage = true
+
+		// check the package state of the application after binary upload
+		var curApp cfapi.CCApp
+		if curApp, err = am.ReadApp(app.ID); err != nil {
+			return
+		}
+		if *curApp.PackageState != "PENDING" {
+			// if it's not already pending, we need to force a restage
+			restage = true
+		} else {
+			// if the package state is pending, we need to restart the application to force an immediate restage
+			// (this is how the CF CLI does this)
```

Comment:
None of the current code in this change came from the CF CLI, and while its behaviour is a good reference, I don't want to use its code because its equivalent operation (cf push) doesn't actually allow for an application's binary to be updated while leaving the application in a stopped state.

Reply:
Thank you for the answer. That's fine then.